plugins/outputs/http/http.go

```diff
@@ -0,0 +1,273 @@
+	}
+
+	return nil
+}
+
+func (h *Http) isExpectedStatusCode(responseStatusCode int) bool {
+	if h.expectedStatusCodeMap == nil {
+		h.expectedStatusCodeMap = make(map[int]bool)
+
+		for _, expectedStatusCode := range h.ExpectedStatusCodes {
+			h.expectedStatusCodeMap[expectedStatusCode] = true
+		}
+	}
+
+	if h.expectedStatusCodeMap[responseStatusCode] {
+		return true
+	}
+
+	return false
+}
+
+// required option validate
+func validate(h *Http) error {
+	if h.URL == "" || len(h.HttpHeaders) == 0 || len(h.ExpectedStatusCodes) == 0 || h.BufferLimit == 0 {
+		return errors.New("E! Http ouput plugin is not working. Because your configuration omits the required option. Please check url, http_headers, expected_status_codes, buffer_limit is empty!")
+	}
+
+	return nil
+}
+
+func (h *Http) write(buf []byte) (*http.Response, error) {
+	req, err := http.NewRequest(POST, h.URL, bytes.NewBuffer(buf))
+
+	for _, httpHeader := range h.HttpHeaders {
+		keyAndValue := strings.Split(httpHeader, ":")
+		req.Header.Set(keyAndValue[0], keyAndValue[1])
+	}
+
+	req.Close = true
+	req.WithContext(h.cancelContext)
+
+	response, err := h.client.Do(req)
+
+	return response, err
+}
+
+// makeRequestBody does additional work so that each serializer can include serialized metrics in the request body.
+// For example, a serialized metric in json.JsonSerializer returns a metric that looks like JsonObject {"key1": "value1"}.
+// In order to accumulate this in the buffer and send it to the request body at once, you need to convert it to Array Json Object [{"key1": "value1"}, {"key2": "value2"}].
+// Thus, makeRequestBody works by making the request body contain the metric returned by each serializer.
+func (h *Http) makeRequestBody() ([]byte, error) {
+	switch h.serializer.(type) {
+	case *json.JsonSerializer:
+		return makeJsonFormatRequestBody(h)
+	default:
+		return makePlainTextFormatRequestBody(h)
+	}
+}
+
+func makePlainTextFormatRequestBody(h *Http) ([]byte, error) {
+	var requestBody bytes.Buffer
+
+	for _, serializedMetric := range h.requestBodyMetricBuffer {
+		requestBody.Write(serializedMetric)
+	}
+
+	return requestBody.Bytes(), nil
+}
+
+func makeJsonFormatRequestBody(h *Http) ([]byte, error) {
+	var requestBody []map[string]interface{}
+
+	for _, serializedMetric := range h.requestBodyMetricBuffer {
+		arrayJsonObject, err := unmarshalArrayJsonObject(serializedMetric)
+
+		if err != nil {
+			fmt.Errorf("E! HTTP json unmarshal is fail! It probably does not seem to fit in the json format. Please check %s", serializedMetric)
+		}
+
+		requestBody = append(requestBody, arrayJsonObject...)
+	}
+
+	h.requestBodyMetricBuffer = nil
+
+	return ejson.Marshal(requestBody)
+}
+
+// unmarshalArrayJsonObject is a case in the Json objects that is Array and there are cases that are not Array Json Object.
+// This is a function that helps to simplify the logic by replacing it with Array.
+func unmarshalArrayJsonObject(buf []byte) ([]map[string]interface{}, error) {
```

Comment:
I needed to make sure each serialized metric was a json object or an array and convert it to an array json object. The reason for consistently converting this to an array json object is to make the logic simpler.
unmarshalArrayJsonObject => convertArrayJsonObject